doc/source/whatsnew/v1.4.0.rst

```diff
@@ -193,6 +193,7 @@ Categorical
 Datetimelike
 ^^^^^^^^^^^^
 - Bug in :class:`DataFrame` constructor unnecessarily copying non-datetimelike 2D object arrays (:issue:`39272`)
+- Bug in :func:`_new_DatetimeIndex` when de-serializing datetime indexes in PYTHONOPTIMIZED mode (:issue:`42866`)
```

Comment:
ok to move to 1,3.2

Reply:
@jreback ok, so I'll have to fork from 1.3.x branch and open a new PR